### PR TITLE
feat: remove auth

### PR DESCRIPTION
**Pull request by acha-bill**
### Checklist

- [x] I have read the [coding guide](https://github.com/ethersphere/bee/blob/master/CODING.md).
- [x] My change requires a documentation update, and I have done it.
- [ ] I have added tests to cover my changes.
- [ ] I have filled out the description and linked the related issues.

### Description
Remove authorization and --restricted option.

Part of #4669

### CODE REVIEW

**Review by janos — COMMENTED** [review]
I think that the "bcrypt" cli command https://github.com/ethersphere/bee/blob/remove-auth/cmd/bee/cmd/bcrypt.go#L15 can be removed, as it was created to help generate admin password hash. I cannot find in the repository that anything else uses bcrypt encryption, but please validate that no other tooling is using this command.

**Review by martinconic — CHANGES_REQUESTED** [review]

File: .github/workflows/beekeeper.yml [hunk 1]
```diff
@@ -14,7 +14,7 @@ env:
   SETUP_CONTRACT_IMAGE: "ethersphere/bee-localchain"
   SETUP_CONTRACT_IMAGE_TAG: "0.8.3"
   BEELOCAL_BRANCH: "main"
-  BEEKEEPER_BRANCH: "master"
+  BEEKEEPER_BRANCH: "api-merge"
```
[the remark this answers is not in the document]
Comment (acha-bill) [reply .github/workflows/beekeeper.yml:17]: yep. I was waiting for beekeeper to be released.

File: pkg/api/api_test.go [hunk 2]
```diff
@@ -230,17 +219,14 @@ func newTestServer(t *testing.T, o testServerOptions) (*http.Client, *websocket.
...
-		Restricted:         o.Restricted,
 	}, extraOpts, 1, erc20)
-	if o.DebugAPI {
-		s.MountTechnicalDebug()
-		s.MountDebug()
-	} else {
-		s.MountAPI()
-	}
+
+	s.MountTechnicalDebug()
```
[the remark this answers is not in the document]
Comment (acha-bill) [reply pkg/api/api_test.go:227]: yes. All the API parts are always mounted

File: pkg/node/node.go [hunk 3]
```diff
@@ -1147,79 +1075,18 @@ func NewBee(
...
-		apiService.Configure(signer, authenticator, tracer, api.Options{
+		apiService.Configure(signer, tracer, api.Options{
 			CORSAllowedOrigins: o.CORSAllowedOrigins,
 			WsPingPeriod:       60 * time.Second,
-			Restricted:         o.Restricted,
 		}, extraOpts, chainID, erc20Service)
 
-		apiService.MountAPI()
 		apiService.MountDebug()
+		apiService.MountAPI()
```
[the remark this answers is not in the document]
Comment (acha-bill) [reply pkg/node/node.go]: Not really, I wanted to be consistent. technical > debug > api

**Review by istae — COMMENTED** [review]
Will to a final review once all the CICD tests are passing.

**Review by martinconic — APPROVED** [review]

**Pull request merged**